pallets/settlement/src/lib.rs

```diff
@@ -458,11 +458,13 @@ decl_module! {
         const MaxLegsInAInstruction: u32 = T::MaxLegsInAInstruction::get();
 
         fn on_runtime_upgrade() -> Weight {
-            // Delete all settlement data
-            let prefix = Twox128::hash(b"Settlement");
-            storage::unhashed::kill_prefix(&prefix);
+            // Delete all settlement data.
+            // This will be a no-op since the data was being stored under a different module hash.
+            storage::unhashed::kill_prefix(&Twox128::hash(b"Settlement"));
+            // Delete data that was erroneously stored under a different module hash.
```

Comment:
Do we only need this for the PMF upgrade (fix), as opposed to the Alcyone upgrade?

Reply:
We need it on Alcyone. Without this fix, the first X instructions after the upgrade on Alcyone will be messed up. X is the number of instructions before the upgrade (currently 1).